frontend/model/state.js

```diff
@@ -44,6 +44,8 @@ setInterval(function () {
   // payments recalculation happen within a minute of day switchover
   if (Math.abs(reactiveDate.date.getTime() - date.getTime()) >= MINS_MILLIS) {
     reactiveDate.date = date
+
+    sbp('gi.actions/group/checkAndUpdateDistributionDate', { contractID: store.state.currentGroupId })
```

Comment:
Please correct me if I'm wrong about anything in below explanations about this change I made.

so, we expect the app to update `distributionDate` of the group as soon as the current period ends and we passes into the new one. but apparently that's not the case currently in the app.

The only place where `distributionDate` is checked and updated is inside `updateCurrentDistribution()` in group.js ([link](https://github.com/okTurtles/group-income/blob/master/frontend/model/contracts/group.js#L96)) and that means the distribution period will remain the old one until this function is triggered by any further user action that runs it, even when the current period actually has ended.
This is a fix for this issue and I encountered this bug while changing around the clock of my machine locally.

`reactiveDate.date` is pretty much the clock that the app uses and every time it gets updated, we check if the current distribution period has ended and if so, we update `distributionDate` accordingly. that way, we can minimise the discrepancy between the current period value and the actual time to less than 1 min.

Hope this sounds as a good improvement and please let me know if there is any additional issue this might bring in. (I haven't encountered any yet.)

Thanks,

Thanks,

Reply:
Hey @SebinSong - the issue is we don't want multiple clients to send this message all at once.

Also, this code I wrote for updating `reactiveDate` is wrong, see #1451.

But, even if #1451 is closed, we still would prefer to do things differently than have all the clients send `checkAndUpdateDistributionDate`, which is what this code currently does.

There are two ways to improve this:

1. Either come up with a scheme where only the first client to correctly notice the new period we're in sends the message.
2. Treat his more like the way we calculate the TODO list for new users when the new period rolls around.

I am in favor of (2) if it is possible (and I'm not sure if it is) because that doesn't require sending any updates to the contract at all. However, figuring out how to do (2) is more difficult. I will think on this in the meantime, but I encourage you to also consider both options. See how we handle the TODO list automatically updating without sending any messages (it's in `groupIncomeAdjustedDistribution` in `state.js`).

Reply:
@taoeffect I used your reference to  `groupIncomeAdjustedDistribution` in `state.js` in the above comment as an inspiration and updated the logic to use `store.watch()` (pls refer to the change in `state.js`) instead of calling the contract action in the setInterval callback. I thought it would make more sense as a watcher. 
But I still haven't figured out how to resolve 
"1. Either come up with a scheme where only the first client to correctly notice the new period we're in sends the message." in here.

let me take more time trying to come up with a better way before/while you review my updated work, but pls let me know if yourself get to find a good solution for it. I will make changes accordingly.

Reply:
Seems like I resolved it by [this commit](https://github.com/okTurtles/group-income/pull/1453/commits/7a79335a32d447ede770eed8b6bf36d42caa6276). it no longer sends multiple messages for `gi.actions/group/updateDistributionDate` when only one is enough. (but like I said, pls let me know if there is a room for improvement re this.)